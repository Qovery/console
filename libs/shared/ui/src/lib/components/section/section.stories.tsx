import type { Meta } from '@storybook/react'
import { Heading } from '../heading/heading'
import { Section } from './section'

const Story: Meta<typeof Section> = {
  component: Section,
  title: 'Section',
  decorators: [
    (Story) => (
      <div style={{ background: 'white', padding: '3em' }}>
        <Story />
      </div>
    ),
  ],
}
export default Story

export const Primary = {
  args: {
    children: (
      <>
        <Heading>Heading 1</Heading>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
        pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est
        laborum.
        <Section>
          <Heading>Heading 2</Heading>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
          consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
          pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
          est laborum.
          <Section>
            <Heading>Heading 3</Heading>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
            dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
            ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
            nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit
            anim id est laborum.
          </Section>
        </Section>
      </>
    ),
  },
}
