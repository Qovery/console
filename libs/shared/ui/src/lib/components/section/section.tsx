import { type ComponentPropsWithoutRef, type ElementRef, forwardRef, useContext } from 'react'
import { twMerge } from '@qovery/shared/util-js'
import { LevelContext } from '../heading/level-context'

export interface SectionProps extends ComponentPropsWithoutRef<'section'> {}

export const Section = forwardRef<ElementRef<'section'>, SectionProps>(function Section(
  { className, children, ...props },
  forwardedRef
) {
  const level = useContext(LevelContext)
  return (
    <section ref={forwardedRef} className={twMerge('flex flex-col', className)} {...props}>
      <LevelContext.Provider value={level + 1}>{children}</LevelContext.Provider>
    </section>
  )
})

export default Section
