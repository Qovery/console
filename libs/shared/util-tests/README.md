# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
