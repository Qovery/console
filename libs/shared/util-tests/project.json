{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
