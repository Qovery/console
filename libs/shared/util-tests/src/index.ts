export * from './lib/shared-util-tests'
