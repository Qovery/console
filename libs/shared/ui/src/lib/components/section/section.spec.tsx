import { renderWithProviders } from '@qovery/shared/util-tests'
import { Section } from './section'

describe('Section', () => {
  it('should match snapshot', () => {
    const { baseElement } = renderWithProviders(<Section>Foobar</Section>)
    expect(baseElement).toMatchSnapshot()
  })
})
