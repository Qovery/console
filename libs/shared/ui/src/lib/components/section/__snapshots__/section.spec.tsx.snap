// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should match snapshot 1`] = `
<body>
  <div>
    <section
      class="flex flex-col"
    >
      Foobar
    </section>
  </div>
</body>
`;
